src/libs/migrations/PersonalDetailsByAccountID.js

```diff
@@ -0,0 +1,208 @@
+import _ from 'underscore';
+import lodashHas from 'lodash/has';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import Log from '../Log';
+
+const DEPRECATED_ONYX_KEYS = {
+    // Deprecated personal details object which was keyed by login instead of accountID.
+    PERSONAL_DETAILS: 'personalDetails',
+};
+
+/**
+ * @returns {Promise<Object>}
+ */
+function getReportActionsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            waitForCollectionCallback: true,
+            callback: (allReportActions) => {
+                Onyx.disconnect(connectionID);
+                return resolve(allReportActions);
+            },
+        });
+    });
+}
+
+/**
+ * @returns {Promise<Object>}
+ */
+function getDeprecatedPersonalDetailsFromOnyx() {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: DEPRECATED_ONYX_KEYS.PERSONAL_DETAILS,
```

Comment:
My thinking is that it's much more efficient to get it from the old object since the old object is keyed by accountID. And if the old object is no longer there, it likely means this migration has already run in the past.

Reply:
Hmm ya I do think it would be nice to explain what you explained in this comment thread just so it's super clear to anyone else looking at this migration in the future 👍